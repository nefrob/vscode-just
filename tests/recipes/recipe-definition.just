# Recipes

# Basic
foo:
  echo 'hello'

abc-DEF-123:
  echo 'hello'

# Parameters

foo param1:
  echo {{param1}}

foo param1 param2:
  echo {{param1}} {{param2}}

foo $bar:
  echo $bar

foo a b=var:
  echo {{a}} {{b}}

default_value a=("a" + ("b" + "c")):
  echo {{a}}

foo a=(arch + "-unknown-unknown") b=(arch / "input.dat"):
  echo {{a}} {{b}}

# TODO: backticks broken
a $A $B=`echo $A`:
  echo $A $B

variadic MESSAGE *FLAGS:
  echo {{MESSAGE}} {{FLAGS}}

plus_variadic +FLAGS='-q':
  echo {{FLAGS}}

# Dependencies

basic: foo
  echo 'bar'

b: a && c d
  echo 'hello from b'

default: (build "main")

complex: lint && (build ("a" + ("b" / "c"))) && test
  echo "hello"
