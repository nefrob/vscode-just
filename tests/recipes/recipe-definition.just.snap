># Recipes
#^^^^^^^^^ source.just comment.line.number-sign.just
>
># Basic
#^^^^^^^ source.just comment.line.number-sign.just
>foo:
#^^^ source.just entity.name.function.just
#   ^ source.just keyword.operator.recipe.end.just
>  echo 'hello'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^^^ source.just string.quoted.single.just
#             ^ source.just string.quoted.single.just
>
>abc-DEF-123:
#^^^^^^^^^^^ source.just entity.name.function.just
#           ^ source.just keyword.operator.recipe.end.just
>  echo 'hello'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^^^ source.just string.quoted.single.just
#             ^ source.just string.quoted.single.just
>
># Parameters
#^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>foo param1:
#^^^ source.just entity.name.function.just
#   ^ source.just
#    ^^^^^^ source.just variable.parameter.recipe.just
#          ^ source.just keyword.operator.recipe.end.just
>  echo {{param1}}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^^^^^^ source.just string.interpolated.escaping.just
#               ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
>foo param1 param2:
#^^^ source.just entity.name.function.just
#   ^ source.just
#    ^^^^^^ source.just variable.parameter.recipe.just
#          ^ source.just
#           ^^^^^^ source.just variable.parameter.recipe.just
#                 ^ source.just keyword.operator.recipe.end.just
>  echo {{param1}} {{param2}}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^^^^^^ source.just string.interpolated.escaping.just
#               ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                 ^ source.just
#                  ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                    ^^^^^^ source.just string.interpolated.escaping.just
#                          ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
>foo $bar:
#^^^ source.just entity.name.function.just
#   ^ source.just
#    ^ source.just keyword.other.recipe.variadic.just
#     ^^^ source.just variable.parameter.recipe.just
#        ^ source.just keyword.operator.recipe.end.just
>  echo $bar
#^^^^^^^^^^^^ source.just
>
>foo a b=var:
#^^^ source.just entity.name.function.just
#   ^ source.just
#    ^ source.just variable.parameter.recipe.just
#     ^ source.just
#      ^ source.just variable.parameter.recipe.just
#       ^ source.just keyword.operator.default.just
#        ^^^ source.just
#           ^ source.just keyword.operator.recipe.end.just
>  echo {{a}} {{b}}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^ source.just string.interpolated.escaping.just
#          ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#            ^ source.just
#             ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#               ^ source.just string.interpolated.escaping.just
#                ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
>default_value a=("a" + ("b" + "c")):
#^^^^^^^^^^^^^ source.just entity.name.function.just
#             ^ source.just
#              ^ source.just variable.parameter.recipe.just
#               ^ source.just keyword.operator.default.just
#                ^ source.just
#                 ^ source.just string.quoted.double.just string.quoted.double.just
#                  ^ source.just string.quoted.double.just
#                   ^ source.just string.quoted.double.just
#                    ^ source.just
#                     ^ source.just keyword.operator.concat.just
#                      ^^ source.just
#                        ^ source.just string.quoted.double.just string.quoted.double.just
#                         ^ source.just string.quoted.double.just
#                          ^ source.just string.quoted.double.just
#                           ^ source.just
#                            ^ source.just keyword.operator.concat.just
#                             ^ source.just
#                              ^ source.just string.quoted.double.just string.quoted.double.just
#                               ^ source.just string.quoted.double.just
#                                ^ source.just string.quoted.double.just
#                                 ^ source.just
#                                  ^ source.just
#                                   ^ source.just keyword.operator.recipe.end.just
>  echo {{a}}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^ source.just string.interpolated.escaping.just
#          ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
>foo a=(arch + "-unknown-unknown") b=(arch / "input.dat"):
#^^^ source.just entity.name.function.just
#   ^ source.just
#    ^ source.just variable.parameter.recipe.just
#     ^ source.just keyword.operator.default.just
#      ^ source.just
#       ^^^^ source.just support.function.builtin.just
#           ^ source.just
#            ^ source.just keyword.operator.concat.just
#             ^ source.just
#              ^ source.just string.quoted.double.just string.quoted.double.just
#               ^^^^^^^^^^^^^^^^ source.just string.quoted.double.just
#                               ^ source.just string.quoted.double.just
#                                ^ source.just
#                                 ^ source.just
#                                  ^ source.just variable.parameter.recipe.just
#                                   ^ source.just keyword.operator.default.just
#                                    ^ source.just
#                                     ^^^^ source.just support.function.builtin.just
#                                         ^ source.just
#                                          ^ source.just keyword.operator.path-join.just
#                                           ^ source.just
#                                            ^ source.just string.quoted.double.just string.quoted.double.just
#                                             ^^^^^^^^^ source.just string.quoted.double.just
#                                                      ^ source.just string.quoted.double.just
#                                                       ^ source.just
#                                                        ^ source.just keyword.operator.recipe.end.just
>  echo {{a}} {{b}}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^ source.just string.interpolated.escaping.just
#          ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#            ^ source.just
#             ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#               ^ source.just string.interpolated.escaping.just
#                ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
># TODO: backticks broken
#^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>a $A $B=`echo $A`:
#^ source.just entity.name.function.just
# ^ source.just
#  ^ source.just keyword.other.recipe.variadic.just
#   ^ source.just variable.parameter.recipe.just
#    ^ source.just
#     ^ source.just keyword.other.recipe.variadic.just
#      ^ source.just variable.parameter.recipe.just
#       ^ source.just keyword.operator.default.just
#        ^ source.just string.interpolated.just
#         ^^^^^^^ source.just source.shell
#                ^ source.just string.interpolated.just
#                 ^ source.just keyword.operator.recipe.end.just
>  echo $A $B
#^^^^^^^^^^^^^ source.just
>
>variadic MESSAGE *FLAGS:
#^^^^^^^^ source.just entity.name.function.just
#        ^ source.just
#         ^^^^^^^ source.just variable.parameter.recipe.just
#                ^ source.just
#                 ^ source.just keyword.other.recipe.variadic.just
#                  ^^^^^ source.just variable.parameter.recipe.just
#                       ^ source.just keyword.operator.recipe.end.just
>  echo {{MESSAGE}} {{FLAGS}}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^^^^^^^ source.just string.interpolated.escaping.just
#                ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                  ^ source.just
#                   ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                     ^^^^^ source.just string.interpolated.escaping.just
#                          ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
>plus_variadic +FLAGS='-q':
#^^^^^^^^^^^^^ source.just entity.name.function.just
#             ^ source.just
#              ^ source.just keyword.other.recipe.variadic.just
#               ^^^^^ source.just variable.parameter.recipe.just
#                    ^ source.just keyword.operator.default.just
#                     ^ source.just string.quoted.single.just string.quoted.single.just
#                      ^^ source.just string.quoted.single.just
#                        ^ source.just string.quoted.single.just
#                         ^ source.just keyword.operator.recipe.end.just
>  echo {{FLAGS}}
#^^^^^^^ source.just
#       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#         ^^^^^ source.just string.interpolated.escaping.just
#              ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
># Dependencies
#^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>basic: foo
#^^^^^ source.just entity.name.function.just
#     ^ source.just keyword.operator.recipe.end.just
#      ^ source.just
#       ^^^ source.just entity.name.function.just
>  echo 'bar'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^ source.just string.quoted.single.just
#           ^ source.just string.quoted.single.just
>
>b: a && c d
#^ source.just entity.name.function.just
# ^ source.just keyword.operator.recipe.end.just
#  ^ source.just
#   ^ source.just entity.name.function.just
#    ^ source.just
#     ^^ source.just keyword.operator.and.just
#       ^ source.just
#        ^ source.just entity.name.function.just
#         ^ source.just
#          ^ source.just entity.name.function.just
>  echo 'hello from b'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^^^^^^^^^^ source.just string.quoted.single.just
#                    ^ source.just string.quoted.single.just
>
>default: (build "main")
#^^^^^^^ source.just entity.name.function.just
#       ^ source.just keyword.operator.recipe.end.just
#        ^ source.just
#         ^ source.just
#          ^^^^^ source.just entity.name.function.just
#               ^ source.just
#                ^ source.just string.quoted.double.just string.quoted.double.just
#                 ^^^^ source.just string.quoted.double.just
#                     ^ source.just string.quoted.double.just
#                      ^ source.just
>
>complex: lint && (build ("a" + ("b" / "c"))) && test
#^^^^^^^ source.just entity.name.function.just
#       ^ source.just keyword.operator.recipe.end.just
#        ^ source.just
#         ^^^^ source.just entity.name.function.just
#             ^ source.just
#              ^^ source.just keyword.operator.and.just
#                ^ source.just
#                 ^ source.just
#                  ^^^^^ source.just entity.name.function.just
#                       ^^ source.just
#                         ^ source.just string.quoted.double.just string.quoted.double.just
#                          ^ source.just string.quoted.double.just
#                           ^ source.just string.quoted.double.just
#                            ^ source.just
#                             ^ source.just keyword.operator.concat.just
#                              ^^ source.just
#                                ^ source.just string.quoted.double.just string.quoted.double.just
#                                 ^ source.just string.quoted.double.just
#                                  ^ source.just string.quoted.double.just
#                                   ^ source.just
#                                    ^ source.just keyword.operator.path-join.just
#                                     ^ source.just
#                                      ^ source.just string.quoted.double.just string.quoted.double.just
#                                       ^ source.just string.quoted.double.just
#                                        ^ source.just string.quoted.double.just
#                                         ^ source.just
#                                          ^ source.just
#                                           ^^ source.just
#                                             ^^ source.just keyword.operator.and.just
#                                               ^ source.just
#                                                ^^^^ source.just entity.name.function.just
>  echo "hello"
#^^^^^^^ source.just
#       ^ source.just string.quoted.double.just string.quoted.double.just
#        ^^^^^ source.just string.quoted.double.just
#             ^ source.just string.quoted.double.just
>